tests/happy_path/test_node-operators-registry-happy-path.py

```diff
@@ -0,0 +1,579 @@
+import pytest
+from web3 import Web3
+from brownie import chain, ZERO_ADDRESS
+from hexbytes import HexBytes
+from typing import NewType, Tuple
+
+from utils.test.extra_data import (
+    ExtraDataService,
+)
+from utils.test.helpers import (
+    shares_balance,
+    ETH,
+    almostEqWithDiff
+)
+from utils.test.oracle_report_helpers import (
+    oracle_report,
+)
+from utils.config import (contracts, lido_dao_staking_router)
+
+PUBKEY_LENGTH = 48
+SIGNATURE_LENGTH = 96
+INITIAL_TOKEN_HOLDER = "0x000000000000000000000000000000000000dead"
+TOTAL_BASIS_POINTS = 10000
+ZERO_HASH = bytes([0] * 32)
+ZERO_BYTES32 = HexBytes(ZERO_HASH)
+
+
+@pytest.fixture()
+def extra_data_service():
+    return ExtraDataService()
+
+
+@pytest.fixture(scope="module")
+def impersonated_voting(accounts):
+    return accounts.at(contracts.voting.address, force=True)
+
+
+StakingModuleId = NewType('StakingModuleId', int)
+NodeOperatorId = NewType('NodeOperatorId', int)
+NodeOperatorGlobalIndex = Tuple[StakingModuleId, NodeOperatorId]
+
+
+def node_operator_gindex(module_id, node_operator_id) -> NodeOperatorGlobalIndex:
+    return module_id, node_operator_id
+
+
+@pytest.fixture(scope="module")
+def nor(interface):
+    return interface.NodeOperatorsRegistry(contracts.node_operators_registry.address)
+
+
+def calc_no_rewards(
```

Comment:
maybe `calc_penalized_nor_rewards` or even `calc_penalized_node_operator_rewards`

Reply:
I think the current name is more appropriate. Since there calculate rewards for any operator, and not just for penalized

Reply:
sorry, but if so, what is the purpose of the following line:
nor_shares = report_shares // 2?

Reply:
Half of the fees are sent to treasury and half to NOR.

Reply:
ah, I see